SyslogNet.Client/Serialization/SyslogRfc5424MessageSerializer.cs

```diff
@@ -33,8 +32,8 @@ public void Serialize(SyslogMessage message, Stream stream)
 
 			writeStream(stream, Encoding.ASCII, messageBuilder.ToString());
 
-			var structuredData = message.StructuredDataElements?.ToList();
-			if (structuredData != null && structuredData.Any())
+			var structuredData = message.StructuredDataElements;

```

Comment:
The ToList is indeed not necessary in this case, it should be removed to avoid a potential costly materialization.